internal/tray-agent/app-indicator/indicator.go

```diff
@@ -247,12 +264,18 @@ func (i *Indicator) SetMenuTitle(title string) {
 
 //Icon returns the icon-id of the Indicator tray icon currently set.
 func (i *Indicator) Icon() Icon {
+	gr := i.graphicResource[resourceIcon]
+	gr.RLock()
+	defer gr.RUnlock()
 	return i.icon
 }
 
 //SetIcon sets the Indicator tray icon. If 'ico' is not a valid argument or ico == IconLiqoNil,
 //SetIcon does nothing.
 func (i *Indicator) SetIcon(ico Icon) {
+	gr := i.graphicResource[resourceIcon]
+	gr.Lock()
```

Comment:
Can we lock just the status write operation instead the whole function body?

Reply:
yes you are right!